src/Microsoft.VisualStudio.ProjectSystem.Managed/IO/Win32FileSystem.cs

```diff
@@ -46,24 +45,18 @@ public void CopyFile(string source, string destination, bool overwrite)
             File.Copy(source, destination, overwrite);
         }
 
-        public string ReadAllText(string path)
+        public async Task<string> ReadAllTextAsync(string path)
         {
-            return File.ReadAllText(path);
+            using var stream = new FileStream(path, FileMode.Open, FileAccess.Read);
+            using var reader = new StreamReader(stream);
+            return await reader.ReadToEndAsync();
         }
 
-        public void WriteAllText(string path, string content)
+        public async Task WriteAllTextAsync(string path, string content)
         {
-            File.WriteAllText(path, content);
-        }
-
-        public void WriteAllText(string path, string content, Encoding encoding)
-        {
-            File.WriteAllText(path, content, encoding);
-        }
-
-        public void WriteAllBytes(string path, byte[] bytes)
-        {
-            File.WriteAllBytes(path, bytes);
+            using var stream = new FileStream(path, FileMode.Create, FileAccess.Write);
```

Comment:
This is a synchronous open and will block the thread; you need to pass isAsync:true.

Reply:
Fixed in #7560.